quickstart-templates/workload-identity-creates-workload-identities/README.md

```diff
@@ -0,0 +1,20 @@
+# Create Workload Identity Federation That Can Create Other Workload Identity Federations
+
+## The Idea
+
+Restrict and limit the need to use high privileges to create resources.
+
+## Objective
+
+A high-privilege user creates an initial workload identity federation token between Azure and a certain GitHub repository. The workload in this repository is then allowed to access Microsoft Entra to create further workload identity federation tokens and assign them to other repositories.
+
+## Benefits
+
+- **Enhanced Security:** Restricting high-privilege usage reduces the attack surface and potential for unauthorized access.
+- **Scalability:** Facilitates the growth of projects by enabling the creation of new identity federations as needed.
+- **Efficient Management:** Streamlines the process of managing resource access across multiple projects and teams.
+
+
+```sh
```

Comment:
Maybe we should start adding a table with least privileged permissions required to run the template for both delegated and app-only modes - showing permissions required to deploy both the ARM resources and the Graph/Entra resources declared in the template.
This could be part of a readme "template" we provide for all such samples...

Reply:
Where could I find this template? Also, how can one systematically ensure the minimum privileges without just blindly trying or "happening to know"? :)

Reply:
This template doesn't exist :(

To **run** the Bicep template that creates the "creator workload identity app", the user would need (I think, but please verify):
- **Entra RBAC permissions**:  [Application Administrator](https://learn.microsoft.com/en-us/entra/identity/role-based-access-control/permissions-reference#application-administrator) and [Privileged Role Administrator](https://learn.microsoft.com/en-us/entra/identity/role-based-access-control/permissions-reference#privileged-role-administrator)
- **Azure RBAC permissions**: [Role-based access control administrator](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/general#role-based-access-control-administrator) and [Managed Identity Contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/identity#managed-identity-contributor) 

The Entra permissions are extremely privileged, especially the 2nd one.  I would advise either that this permissions are temporarily assigned for the purpose of just running the bootstrap Bicep template, or are assigned via Privileged Identity Management (if the tenant has Entra licenses/subscriptions for this).

This Bicep template itself needs to assign permissions to the creator workload identity (to create other apps, to create managed identities and assign Azure roles) and you can describe those in the template's comments.